samza-kafka/src/main/scala/org/apache/samza/system/kafka_deprecated/KafkaSystemConsumer.scala

```diff
@@ -284,12 +284,12 @@ private[kafka_deprecated] class KafkaSystemConsumer(
       }
 
       if(fetchLimitByBytesEnabled ) {
-        val ime = new IncomingMessageEnvelope(systemStreamPartition, offset, key, message, getMessageSize(msg.message))
-        ime.setTimestamp(if (!msg.message.isNull) msg.message.timestamp else 0L)
+        val ime = new IncomingMessageEnvelope(systemStreamPartition, offset, key, message, getMessageSize(msg.message),
+          (if (!msg.message.isNull) msg.message.timestamp else 0L), 0L)
```

Comment:
Extract as a field outside the if(fetchLimitByBytesEnabled) block.
Also, should we set arrivalTime to currentTimeMilis here?